device/iot-device-samples/device-reconnection-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/IotHubConnectionStatusChangeCallbackHandler.java

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft. All rights reserved.Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+package samples.com.microsoft.azure.sdk.iot;
+
+import com.microsoft.azure.sdk.iot.device.IotHubConnectionStatusChangeCallback;
+import com.microsoft.azure.sdk.iot.device.IotHubConnectionStatusChangeReason;
+import com.microsoft.azure.sdk.iot.device.transport.IotHubConnectionStatus;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class IotHubConnectionStatusChangeCallbackHandler implements IotHubConnectionStatusChangeCallback {
+    @Override
+    public void execute(IotHubConnectionStatus status, IotHubConnectionStatusChangeReason statusChangeReason, Throwable throwable, Object callbackContext) {
+        log.debug("### Connection status change reported: status={}, reason={}", status, statusChangeReason);
+
+        if (throwable != null) {
+            log.debug("Throwable: {}", throwable.getMessage());
+            throwable.printStackTrace();
```

Comment:
log.warn("Connection status callback executed with a throwable");
log.warn("Throwable: {}", throwable);

Reply:
.printStackTrace() won't necessarily log to the same place as slf4j, so we should make sure both messages are logged to slf4j ourselves

---
In reply to: [339193139](https://github.com/Azure/azure-iot-sdk-java/pull/612#discussion_r339193139) [](ancestors = 339193139)

Reply:
good point; i'll make the change